### PR TITLE
enumerates the states in which code is executed...

### PR DESCRIPTION
**Pull request by zevaverbach**
... and fixes an awkward phrasing.

---

This change is [Reviewable](https://reviewable.io/reviews/pallets/flask/1887)

### CODE REVIEW

File: docs/appcontext.rst [hunk 1]
```diff
@@ -5,31 +5,37 @@ The Application Context
...
-:class:`Flask` object is instantiated, and it implicitly ends when the
-first request comes in.  While the application is in this state a few
-assumptions are true:
+One of the design ideas behind Flask is that there are at least two 
+different “states” in which code is executed:
+
+1. The application setup state, in which the application implicitly is 
+on the module level.  
+
+This state starts when the :class:`Flask` object is instantiated, and
```
Comment (untitaker) [inline docs/appcontext.rst]: This needs to be indented such that it is part of the list item.

Comment (zevav) [inline docs/appcontext.rst]: I think I just fixed this [here](https://github.com/pallets/flask/pull/1887/commits/863e5cca1b498fb3170981892e820df97f4200e7) .

File: docs/appcontext.rst [hunk 2]
```diff
@@ -5,31 +5,37 @@ The Application Context
...
+on the module level.  
+
+    This state starts when the :class:`Flask` object is instantiated, and 
+    it implicitly ends when the first request comes in.  While the 
+    application is in this state, a few assumptions are true:
+
+    -   the programmer can modify the application object safely.
+    -   no request handling happened so far
+    -   you have to have a reference to the application object in order to
+        modify it, there is no magic proxy that can give you a reference to
```
Comment (davidism) [inline docs/appcontext.rst]: Instead (or in addition to) "magic proxy", should probably just say something along the lines of "`current_app` will not point to the app yet".

File: docs/appcontext.rst [hunk 3]
```diff
@@ -5,31 +5,37 @@ The Application Context
...
-    the application object you're currently creating or modifying.
-
-In contrast, during request handling, a couple of other rules exist:
-
--   while a request is active, the context local objects
+One of the design ideas behind Flask is that there are at least two 
+different “states” in which code is executed:
+
+1.  The application setup state, in which the application implicitly is 
+on the module level.
```
Comment (davidism) [inline docs/appcontext.rst]: I'm not sure what "in which the application implicitly is on the module level" was originally trying to say.  It makes even less sense when using the factory pattern too, when the app is only accessible in the function scope.  I think this should be removed, the real information about this state is given below.

File: docs/appcontext.rst [hunk 4]
```diff
@@ -5,31 +5,37 @@ The Application Context
...
+    it implicitly ends when the first request comes in.  While the 
+    application is in this state, a few assumptions are true:
+
+    -   the programmer can modify the application object safely.
+    -   no request handling happened so far
+    -   you have to have a reference to the application object in order to
+        modify it, there is no magic proxy that can give you a reference to
+        the application object you're currently creating or modifying.
+
+2.  In contrast, in the request handling state, a couple of other rules
```
Comment (davidism) [inline docs/appcontext.rst]: Make this just "The request handling state:"

File: docs/appcontext.rst [hunk 5]
```diff
@@ -5,31 +5,37 @@ The Application Context
...
-
-There is a third state which is sitting in between a little bit.
-Sometimes you are dealing with an application in a way that is similar to
-how you interact with applications during request handling; just that there
-is no request active.  Consider, for instance, that you're sitting in an
-interactive Python shell and interacting with the application, or a
-command line application.
+    -   any code can get hold of these objects at any time.
+
+3.  There is also a third state somewhere in between 'module-level' and
```
Comment (davidism) [inline docs/appcontext.rst]: "in between the application setup and request handling states"

Comment (zevav) [inline docs/appcontext.rst]: Do you want me to have a crack at implementing these notes?  Just want to be sure.

Comment (untitaker) [inline docs/appcontext.rst]: @zevav Yes I think that's the intention. Keep in mind that this is all voluntary though.

Comment (zevav) [inline docs/appcontext.rst]: cool, i will do so: have to get a little bit more into the internals; should I throw questions at you guys here or on IRC?

Comment (untitaker) [inline docs/appcontext.rst]: Probably IRC but you might not get a response there due to timezone differences.

**Comment by davidism** [comment]
You can come chat at http://chat.stackoverflow.com/rooms/6/python if you have questions, just ping me there.

File: docs/appcontext.rst (hunk 3 above)
Comment (untitaker) [inline docs/appcontext.rst]: I think those reviews should be part of a new PR, this is just about enumerating

**Comment by kennethreitz** [comment]
✨🍰✨

**Pull request merged**